okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -150,39 +157,39 @@ public void onFailure(Call call, IOException exception) {
             }
 
             @Override
-            public void onResponse(Call call, Response unbufferedResponse) throws IOException {
+            public void onResponse(Call call, Response response) throws IOException {
                 // Relay successful responses
-                if (unbufferedResponse.code() / 100 <= 2) {
-                    callback.onResponse(call, unbufferedResponse);
+                if (response.code() / 100 <= 2) {
+                    callback.onResponse(call, response);
                     return;
                 }
 
                 // Buffer the response into a byte[] so that multiple handler can safely consume the body.
-                // The buffer call consumes and closes the original unbufferedResponse body.
-                final Response response;
+                // The buffer call consumes and closes the original response body.
+                byte[] bodyBytes;
```

Comment:
instead of forcing the downstreams to deal with the notion of response and its body, maybe just have something like

```
Supplier<Response> responseSupplier;
try {
    byte[] body = response.body().bytes();
    // so handlers can read the body
    responseSupplier = () -> buildFrom(response, bodyBytes);
}
```
?

and then it's just responseSupplier.get()?